Implementation Of The Single Proposal Consultation For A Member

alexisljn opened the pull request:
Add of a UrlManager rule in config/web.
Creation of 2 methods :
1 to check if the proposal requested exists.
1 to check if the requester is the owner of the proposal.

alexisljn committed:
Implementation Of The Single Proposal Consultation For A Member

Add of a UrlManager rule in config/web.
Creation of 2 methods :
1 to check if the proposal requested exists.
1 to check if the requester is the owner of the proposal.

alexisljn committed:
Generation Of The Chronological Stream Of A Proposal

Takes comments, reviews and old content to merge them into an array.

alexisljn committed:
Sort Of The Chronological Streeam

Sort by date to get similar behavior than gitHub stream.

alexisljn committed:
Display Of Proposal Informations In View

Title, creation date, last edit date, status, social media and content.

alexisljn committed:
Display In View Of All Needed Data About A Proposal

It means the proposal itself, its history, its comments
and its reviews.

alexisljn commented:
Look at the video to see the feature with at least 1 old content for a proposal. 

https://www.youtube.com/watch?v=EEOQaiEX60k&feature=youtu.be

alexisljn committed:
Removal Of Web Config Comment

In order to keep this file clean.

alexisljn committed:
Encoding Of User Input In View Output

To prevent the XSS attack.

EmileCalixte commented on controllers/ProposalController.php:
```diff
@@ -27,6 +55,91 @@ public function actionMyProposals(): string
+        if($submitterId == self::getCurrentUser()->id) {
+            return;
+        }
+        throw new Exception('Not owner of this proposal');
```
see line 13 : the Exception you imported is the yii\db\Exception. You should throw an UnauthorizedException, or simply return false as it's a private function

alexisljn replied:
It's a temporary Exception waiting for creating a customized exception for that case. I make the change in case we haven't time to design it.

EmileCalixte commented on controllers/ProposalController.php:
```diff
@@ -5,19 +5,47 @@
+                $selectedProposal->reviews,
+                $selectedProposal->proposalContentHistories
+            );
+            $chronologicalStream = $this->sortChronologicalStreamByDate($chronologicalStream);
```
Why not just generate a sorted chronological stream with the generateChronologicalStream function ?

alexisljn replied:
You're right, a not chronological stream is never needed

alexisljn committed:
Correction For Approval Of PR

Use of Unauthorized exception instead of Exception.
Factorization of methods about the chronological stream.
